### PR TITLE
doctests for calling c and fortran

### PR DESCRIPTION
**Pull request by KristofferC**
Some tests on the few places I could squeeze them in

Edit: canceled CI

### CODE REVIEW

File: doc/src/manual/calling-c-and-fortran-code.md [hunk 1]
```diff
@@ -853,7 +867,7 @@ macro dlsym(func, lib)
     end
```
Comment (tkelman) [inline doc/src/manual/calling-c-and-fortran-code.md]: there's some inconsistent indentation in here

Comment (KristofferC) [inline doc/src/manual/calling-c-and-fortran-code.md]: I cant see it

Comment (KristofferC) [inline doc/src/manual/calling-c-and-fortran-code.md]: I updated some other indentation problems I found though

Comment (tkelman) [inline doc/src/manual/calling-c-and-fortran-code.md]: it was collapsed and this was the closest line github let me comment on, I think you got it

Comment (tkelman) [inline doc/src/manual/calling-c-and-fortran-code.md]: github moved my comment to a different line again. I was talking about the dlsym macro definition

**Pull request merged**